### PR TITLE
Fix Vagrant inbound connections

### PR DESCRIPTION
**Pull request by kklin**
This PR fixes inbound connections on Vagrant by allowing the Vagrant provider to override the interface that NAT rules should be applied to. While in the cluster code, I cleaned up various things about the `machine.Machine` struct to make it cleaner.

### CODE REVIEW

File: cluster/amazon/amazon.go [hunk 1]
```diff
@@ -10,7 +10,6 @@ import (
 	"github.com/quilt/quilt/cluster/acl"
```
Comment (ejj) [inline cluster/amazon/amazon.go]: In the commit message, did you mean "these attribute are unknown by the provider"

File: cluster/cluster.go [hunk 2]
```diff
@@ -217,13 +217,21 @@ func (clst cluster) updateCloud(machines []machine.Machine, act action) {
 	}
 }
 
+type joinMachine struct {
```
Comment (ejj) [inline cluster/cluster.go]: I think I'd just call it `machine`

File: cluster/amazon/amazon.go (hunk 1 above)
Comment (kklin) [inline cluster/amazon/amazon.go]: Thanks, I meant "These attributes are known by the cluster"

File: cluster/cluster.go (hunk 2 above)
Comment (kklin) [inline cluster/cluster.go]: That conflicts with the package `github.com/quilt/quilt/cluster/machine` unfortunately.

**Pull request merged**